src/main/java/com/idrsolutions/microservice/BaseServletContextListener.java

```diff
@@ -241,4 +243,18 @@ private void validateFileDeletionServiceFrequency(final Properties properties) {
             LOG.log(Level.WARNING, message);
         }
     }
+
+    private static void validateMaxConversionDuration(final Properties properties) {
+        final String maxDuration = properties.getProperty(KEY_PROPERTY_MAX_CONVERSION_DURATION);
+        if (maxDuration == null || maxDuration.isEmpty() || "0".equals(maxDuration) || !(maxDuration.matches("\\d+") || "Infinity".equals(maxDuration))) {
+            properties.setProperty(KEY_PROPERTY_MAX_CONVERSION_DURATION, String.valueOf(Long.MAX_VALUE));
+            final String message = String.format("Properties value for \"maxConversionDuration\" was set to " +
+                    "\"%s\" but should be a positive long or Infinity. Using a value of Infinity.", maxDuration);
+            LOG.log(Level.WARNING, message);
+        } else if ("Infinity".equals(maxDuration)) {
+            properties.setProperty(KEY_PROPERTY_MAX_CONVERSION_DURATION, String.valueOf(Long.MAX_VALUE));
+            final String message = String.format("Interpreting \"Infinity\" as %d", Long.MAX_VALUE);
+            LOG.log(Level.INFO, message);
```

Comment:
It looks like this code is unreachable. If `"Infinity".equals(maxDuration)` is true then it would be captured by the first branch of the if statement?

Reply:
The `"Infinity".equals(maxDuration)` in line 249 is actually negated because it's part of `!(maxDuration.matches("\\d+") || "Infinity".equals(maxDuration))`.

I should probably put that on a new line to make it clearer.

Reply:
I have changed the if conditions to make it easier to read. The NOT(A OR B) is now a (NOT A AND NOT B) which is logically identical. It is now also on a separate line.

Reply:
My mistake! Nice change though, I think it's clearer now